source/extensions/filters/http/adaptive_concurrency/concurrency_controller/gradient_controller.h

```diff
@@ -70,36 +71,39 @@ class GradientControllerConfig : public Logger::Loggable<Logger::Id::filter> {
   }
 
   double maxGradient() const {
-    return runtime_.snapshot().getDouble(RuntimeKeys::get().MaxGradientKey, max_gradient_);
+    return std::max(
+        1.0, runtime_.snapshot().getDouble(RuntimeKeys::get().MaxGradientKey, max_gradient_));
   }
 
   // The percentage is normalized to the range [0.0, 1.0].
   double sampleAggregatePercentile() const {
-    return runtime_.snapshot().getDouble(RuntimeKeys::get().SampleAggregatePercentileKey,
-                                         sample_aggregate_percentile_) /
-           100.0;
+    const double val = runtime_.snapshot().getDouble(
```

Comment:
Nit: prefer code and major docs PR to be separate, but it's fine for this PR.

Reply:
+1 in the future.